src/main/java/org/mastodon/revised/mamut/feature/LinkVelocityFeatureComputer.java

```diff
@@ -0,0 +1,68 @@
+package org.mastodon.revised.mamut.feature;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.mastodon.properties.DoublePropertyMap;
+import org.mastodon.revised.model.feature.Feature;
+import org.mastodon.revised.model.feature.FeatureProjection;
+import org.mastodon.revised.model.feature.FeatureProjectors;
+import org.mastodon.revised.model.mamut.Link;
+import org.mastodon.revised.model.mamut.Model;
+import org.mastodon.revised.model.mamut.ModelGraph;
+import org.mastodon.revised.model.mamut.Spot;
+import org.scijava.plugin.Plugin;
+
+@Plugin( type = LinkFeatureComputer.class, name = "Link velocity" )
+public class LinkVelocityFeatureComputer implements LinkFeatureComputer
+{
+
+	public static final String KEY = "Link velocity";
+
+	@Override
+	public Set< String > getDependencies()
+	{
+		return Collections.singleton( LinkDisplacementComputer.KEY );
+	}
+
+	@Override
+	public String getKey()
+	{
+		return KEY;
+	}
+
+	@Override
+	public Feature< Link, Double, DoublePropertyMap< Link > > compute( final Model model )
+	{
+		final ModelGraph graph = model.getGraph();
+		final DoublePropertyMap< Link > pm = new DoublePropertyMap<>( graph.edges(), Double.NaN );
+
+		final Spot ref1 = graph.vertexRef();
+		final Spot ref2 = graph.vertexRef();
+
+		@SuppressWarnings( "unchecked" )
+		final Feature< Link, Double, DoublePropertyMap< Link > > displacementFeature =
+				( Feature< Link, Double, DoublePropertyMap< Link > > ) model.getFeatureModel().getFeature( LinkDisplacementComputer.KEY );
```

Comment:
Would it be possible to make `FeatureModel.getFeature(...)` trigger computation if the feature is not present?
Might require to fuse `DefaultFeatureModel` and `DefaultFeatureComputerService`. This would provide automatic dependency resolution. 
(In any case, I would additionally keep to the explicit dependency resolution topological sort for the progress report and circular dependency detection.)

Reply:
I need time to think about this.

Reply:
I am not sure I feel great about going this direction.
I feel like the calculation and the accessing of features should be separated. Accessing a feature value should not trigger possibly crazy calculation, because this is the user responsibility to trigger calculation. We have the logic to inform the user that a feature value is not available for the object he is trying to access, and it his their responsibility to trigger calculation.